util/cache/cache.go

```diff
@@ -0,0 +1,57 @@
+package cache
+
+import (
+	"sync"
+	"time"
+)
+
+type cachedValue[T any] struct {
+	value     T
+	expiresAt time.Time
+}
+
+func (c *cachedValue[T]) isExpired(lifetime time.Duration) bool {
+	return time.Until(c.expiresAt) < lifetime
+}
+
+type cache[T any] struct {
+	cache map[string]cachedValue[T]
+	mu    sync.RWMutex
+}
+
+// New creates a new in-memory cache.
+func New[T any]() *cache[T] {
```

Comment:
It's more idiomatic to return exported types from exported functions.

Reply:
That checks!